backend/app/api/endpoints/knowledge.py

```
@@ -80,14 +129,16 @@ def list_knowledge_bases(
+    - **scope=organization**: Organization-level knowledge bases (accessible to all users)
+    - **scope=external**: Knowledge bases the user has been granted access to (via explicit permission)
     - **scope=all**: All accessible knowledge bases (personal + team)
     """
     try:
         resource_scope = ResourceScope(scope)
     except ValueError:
         raise HTTPException(
             status_code=status.HTTP_400_BAD_REQUEST,
-            detail=f"Invalid scope: {scope}. Must be one of: personal, group, all",
+            detail=f"Invalid scope: {scope}. Must be one of: personal, group, external, all",
         )
```

Ruff (0.14.14)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)